Enable KubeArmor Runtime Security Support on GKE Autopilot (DaemonSet/Privileged Pod Restrictions)

## Feature Request

Support Native WorkloadAllowlist CR Deployment in Clusters

kubearmor will require exemptions for some of the default security measures enforced in autopilot mode of a GKE cluster. Two workloads, `kubearmor daemonset` and `kubearmor-snitch` will require exemption to use certain linux capabilities and host path mount that are not allowed in autopilot mode.
 
### exemptions:

```
- autogke-disallow-hostnamespaces  <-- for kubearmor daemonset only
- autogke-default-linux-capabilities
- autogke-no-write-mode-hostpath
```
### deployment limitation

major blocker with operator deployment model is that by default direct node operations are blocked by design in autopilot and due to this `kubearmor-snitch` will not be able to patch the nodes with labels that are required to dynamically generate the `kubearmor daemonset`.

```
{"level":"info","ts":1756465664.0644646,"caller":"k8s/client.go:21","msg":"Trying to load InCluster configuration"}
{"level":"info","ts":1756465664.065652,"caller":"k8s/client.go:39","msg":"Loaded InCluster configuration"}
{"level":"info","ts":1756465664.0857096,"caller":"snitch-cmd/main.go:72","msg":"Running snitch in node gk3-ramakant-cluster-nap-1cgxn2f3-dda9eac8-q2tg"}
{"level":"info","ts":1756465664.0857918,"caller":"snitch-cmd/main.go:73","msg":"lsm order=bpf,apparmor,selinux"}
{"level":"info","ts":1756465664.0858014,"caller":"snitch-cmd/main.go:74","msg":"path prefix=/rootfs/"}
{"level":"info","ts":1756465664.0858078,"caller":"snitch-cmd/main.go:75","msg":"k8s runtime=containerd"}
{"level":"info","ts":1756465664.0858247,"caller":"snitch-cmd/main.go:76","msg":"KubeConfig path=/root/.kube/config"}
{"level":"info","ts":1756465664.0871155,"caller":"enforcer/enforcer.go:94","msg":"/sys/kernel/security/lsm : capability,lockdown,yama,loadpin,safesetid,apparmor,bpf,csm"}
{"level":"info","ts":1756465664.0871701,"caller":"enforcer/enforcer.go:95","msg":"Supported LSMs capability,lockdown,yama,loadpin,safesetid,apparmor,bpf,csm"}
{"level":"info","ts":1756465664.0871785,"caller":"snitch-cmd/main.go:129","msg":"Node enforcer is bpf"}
{"level":"info","ts":1756465664.0871985,"caller":"runtime/runtime.go:29","msg":"Checking for containerd socket\n"}
{"level":"warn","ts":1756465664.0872245,"caller":"runtime/runtime.go:42","msg":"stat /rootfs//var/snap/microk8s/common/run/containerd.sock: no such file or directory"}
{"level":"warn","ts":1756465664.0872495,"caller":"runtime/runtime.go:42","msg":"stat /rootfs//run/k0s/containerd.sock: no such file or directory"}
{"level":"warn","ts":1756465664.0872743,"caller":"runtime/runtime.go:42","msg":"stat /rootfs//run/k3s/containerd/containerd.sock: no such file or directory"}
2025-08-29 11:07:44.087497      ESC[33mWARNESC[0m       stat /rootfs/var/run/nri/nri.sock: no such file or directory
{"level":"info","ts":1756465664.0880544,"caller":"snitch-cmd/main.go:138","msg":"Detected containerd as node runtime, runtime socket=/run/containerd/containerd.sock"}
{"level":"info","ts":1756465664.0880747,"caller":"snitch-cmd/main.go:164","msg":"Kernel has BTF: yes"}
{"level":"error","ts":1756465664.1441495,"caller":"snitch-cmd/main.go:194","msg":"Error while patching node gk3-ramakant-cluster-nap-1cgxn2f3-dda9eac8-q2tg error=admission webhook \"warden-validating.common-webhooks.networking.gke.io\" denied the request: GKE Warden rejected the request because it violates one or more constraints.\nViolations details: {\"[denied by autogke-no-node-updates]\":[\"Operation on Autopilot nodes with changes in addition to cordon is not allowed.\"]}\nRequested by user: 'system:serviceaccount:kubearmor:kubearmor-snitch', groups: 'system:serviceaccounts,system:serviceaccounts:kubearmor,system:authenticated'.","stacktrace":"main.snitch\n\t/KubeArmor/pkg/KubeArmorOperator/cmd/snitch-cmd/main.go:194\nmain.init.func2\n\t/KubeArmor/pkg/KubeArmorOperator/cmd/snitch-cmd/main.go:77\ngithub.com/spf13/cobra.(*Command).execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1019\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148\ngithub.com/spf13/cobra.(*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071\nmain.Execute\n\t/KubeArmor/pkg/KubeArmorOperator/cmd/snitch-cmd/main.go:115\nmain.main\n\t/KubeArmor/pkg/KubeArmorOperator/cmd/snitch-cmd/main.go:260\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:283"}
```
### functional challenge

so to avoid this limitation we're working with predefined set of configuration for `kubearmor daemonset`, by default nodes in autopilot mode configured with Container-Optimized OS with containerd as the node operating system. considering this a custom created daemonset was deployed but faced some issues working with bpf filesystem.

```
2025-08-29 12:23:13.915449	[34mINFO[0m	BUILD-INFO: version: v1.6.3-dirty, branch: HEAD, date: 2025-08-19T18:03:25Z
2025-08-29 12:23:13.917218	[34mINFO[0m	Arguments [alertThrottling:true annotateResources:false bpfFsPath:/sys/fs/bpf cluster:default coverageTest:false criSocket: debug:false defaultCapabilitiesPosture:audit defaultFilePosture:audit defaultNetworkPosture:audit defaultPostureLogs:true enableKubeArmorHostPolicy:false enableKubeArmorPolicy:true enableKubeArmorStateAgent:false enableKubeArmorVm:false enableNRI:false enforcerAlerts:true gRPC:32767 host:gk3-ramakant-cluster-nap-1cgxn2f3-06f5f5a2-j748 hostDefaultCapabilitiesPosture:audit hostDefaultFilePosture:audit hostDefaultNetworkPosture:audit hostVisibility:default initTimeout:60s k8s:true kubeconfig: logPath:none lsm:bpf,apparmor,selinux machineIDPath:/etc/machine-id maxAlertPerSec:10 nriIndex:99 nriSocket: procfsMount:/proc seLinuxProfileDir:/tmp/kubearmor.selinux throttleSec:30 tlsCertPath:/var/lib/kubearmor/tls tlsCertProvider:self tlsEnabled:false untrackedNs:kube-system,kubearmor useOCIHooks:false visibility:process,file,network,capabilities]
2025-08-29 12:23:13.917682	[34mINFO[0m	Configuration [{Cluster: Host: GRPC: TLSEnabled:false TLSCertPath: TLSCertProvider: LogPath: SELinuxProfileDir: CRISocket: NRISocket: NRIIndex: NRIEnabled:false Visibility: HostVisibility: Policy:false HostPolicy:false KVMAgent:false K8sEnv:false Debug:false DefaultFilePosture: DefaultNetworkPosture: DefaultCapabilitiesPosture: HostDefaultFilePosture: HostDefaultNetworkPosture: HostDefaultCapabilitiesPosture: CoverageTest:false ConfigUntrackedNs:[] LsmOrder:[] BPFFsPath: EnforcerAlerts:false DefaultPostureLogs:false InitTimeout: StateAgent:false UseOCIHooks:false AlertThrottling:false MaxAlertPerSec:0 ThrottleSec:0 AnnotateResources:false ProcFsMount: MachineIDPath:}]
2025-08-29 12:23:13.917801	[34mINFO[0m	Final Configuration [{Cluster:default Host:gk3-ramakant-cluster-nap-1cgxn2f3-06f5f5a2-j748 GRPC:32767 TLSEnabled:false TLSCertPath:/var/lib/kubearmor/tls TLSCertProvider:self LogPath:none SELinuxProfileDir: CRISocket: NRISocket: NRIIndex:99 NRIEnabled:false Visibility:process,file,network,capabilities HostVisibility:none Policy:true HostPolicy:false KVMAgent:false K8sEnv:true Debug:false DefaultFilePosture:audit DefaultNetworkPosture:audit DefaultCapabilitiesPosture:audit HostDefaultFilePosture:audit HostDefaultNetworkPosture:audit HostDefaultCapabilitiesPosture:audit CoverageTest:false ConfigUntrackedNs:[kube-system kubearmor] LsmOrder:[bpf apparmor selinux] BPFFsPath:/sys/fs/bpf EnforcerAlerts:true DefaultPostureLogs:true InitTimeout:60s StateAgent:false UseOCIHooks:false AlertThrottling:true MaxAlertPerSec:10 ThrottleSec:30 AnnotateResources:false ProcFsMount:/host/procfs MachineIDPath:/etc/machine-id}]
2025-08-29 12:23:13.917854	[34mINFO[0m	Final Configuration [{Cluster:default Host:gk3-ramakant-cluster-nap-1cgxn2f3-06f5f5a2-j748 GRPC:32767 TLSEnabled:false TLSCertPath:/var/lib/kubearmor/tls TLSCertProvider:self LogPath:none SELinuxProfileDir: CRISocket: NRISocket: NRIIndex:99 NRIEnabled:false Visibility:process,file,network,capabilities HostVisibility:none Policy:true HostPolicy:false KVMAgent:false K8sEnv:true Debug:false DefaultFilePosture:audit DefaultNetworkPosture:audit DefaultCapabilitiesPosture:audit HostDefaultFilePosture:audit HostDefaultNetworkPosture:audit HostDefaultCapabilitiesPosture:audit CoverageTest:false ConfigUntrackedNs:[kube-system kubearmor] LsmOrder:[bpf apparmor selinux] BPFFsPath:/sys/fs/bpf EnforcerAlerts:true DefaultPostureLogs:true InitTimeout:60s StateAgent:false UseOCIHooks:false AlertThrottling:true MaxAlertPerSec:10 ThrottleSec:30 AnnotateResources:false ProcFsMount:/host/procfs MachineIDPath:/etc/machine-id}]
2025-08-29 12:23:13.919608	[34mINFO[0m	Initialized Kubernetes client
2025-08-29 12:23:13.919661	[34mINFO[0m	Started to monitor node events
2025-08-29 12:23:13.919704	[34mINFO[0m	GlobalCfg.Host=gk3-ramakant-cluster-nap-1cgxn2f3-06f5f5a2-j748, KUBEARMOR_NODENAME=gk3-ramakant-cluster-nap-1cgxn2f3-06f5f5a2-j748
2025-08-29 12:23:13.919782	[34mINFO[0m	Started watching node information
2025-08-29 12:23:13.932483	[31mERROR[0m	Failed to read /sys/kernel/security/lsm: empty file
github.com/kubearmor/KubeArmor/KubeArmor/log.Err
	/usr/src/KubeArmor/KubeArmor/log/logger.go:103
github.com/kubearmor/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).HandleNodeAnnotations
	/usr/src/KubeArmor/KubeArmor/core/kubeUpdate.go:71
github.com/kubearmor/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).checkAndUpdateNode
	/usr/src/KubeArmor/KubeArmor/core/kubeUpdate.go:161
github.com/kubearmor/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).WatchK8sNodes.func2
	/usr/src/KubeArmor/KubeArmor/core/kubeUpdate.go:190
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd
	/go/pkg/mod/k8s.io/client-go@v0.33.3/tools/cache/controller.go:257
k8s.io/client-go/tools/cache.(*processorListener).run.func1
	/go/pkg/mod/k8s.io/client-go@v0.33.3/tools/cache/shared_informer.go:1077
k8s.io/client-go/tools/cache.(*processorListener).run
	/go/pkg/mod/k8s.io/client-go@v0.33.3/tools/cache/shared_informer.go:1087
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.33.3/pkg/util/wait/wait.go:72
2025-08-29 12:23:14.920694	[34mINFO[0m	Node Name: gk3-ramakant-cluster-nap-1cgxn2f3-06f5f5a2-j748
2025-08-29 12:23:14.920757	[34mINFO[0m	Node IP: 10.128.15.203
2025-08-29 12:23:14.920777	[34mINFO[0m	Node ID: 3c37a17d6053ec522fffeb49f599721b3151bd89d5adbfb333dd998c75d64b6e
2025-08-29 12:23:14.920834	[34mINFO[0m	Node Annotations: map[container.googleapis.com/instance_id:4669353774271967103 csi.volume.kubernetes.io/nodeid:{"filestore.csi.storage.gke.io":"gk3-ramakant-cluster-nap-1cgxn2f3-06f5f5a2-j748","gcsfuse.csi.storage.gke.io":"gk3-ramakant-cluster-nap-1cgxn2f3-06f5f5a2-j748","parallelstore.csi.storage.gke.io":"gk3-ramakant-cluster-nap-1cgxn2f3-06f5f5a2-j748","pd.csi.storage.gke.io":"projects/accuknox-cnapp/zones/us-central1-a/instances/gk3-ramakant-cluster-nap-1cgxn2f3-06f5f5a2-j748","podcertificate.gke.io":"gk3-ramakant-cluster-nap-1cgxn2f3-06f5f5a2-j748"} kubearmor-policy:audited kubearmor-visibility:none node.alpha.kubernetes.io/ttl:0 node.gke.io/last-applied-node-labels:addon.gke.io/node-local-dns-ds-ready=true,cloud.google.com/gke-boot-disk=pd-balanced,cloud.google.com/gke-container-runtime=containerd,cloud.google.com/gke-cpu-scaling-level=16,cloud.google.com/gke-gcfs=true,cloud.google.com/gke-image-streaming=true,cloud.google.com/gke-logging-variant=DEFAULT,cloud.google.com/gke-max-pods-per-node=64,cloud.google.com/gke-memory-gb-scaling-level=65,cloud.google.com/gke-netd-ready=true,cloud.google.com/gke-nodepool=nap-1cgxn2f3,cloud.google.com/gke-os-distribution=cos,cloud.google.com/gke-provisioning=standard,cloud.google.com/gke-stack-type=IPV4,cloud.google.com/machine-family=ek,cloud.google.com/private-node=false,iam.gke.io/gke-metadata-server-enabled=true,node.kubernetes.io/masq-agent-ds-ready=true node.gke.io/last-applied-node-taints: volumes.kubernetes.io/controller-managed-attach-detach:true]
2025-08-29 12:23:14.920848	[34mINFO[0m	OS Image: Container-Optimized OS from Google
2025-08-29 12:23:14.920864	[34mINFO[0m	Kernel Version: 6.6.93+
2025-08-29 12:23:14.920882	[34mINFO[0m	Kubelet Version: v1.33.2-gke.1240000
2025-08-29 12:23:14.920900	[34mINFO[0m	Container Runtime: containerd://2.0.4
2025-08-29 12:23:14.926588	[34mINFO[0m	Initialized KubeArmor Logger
2025-08-29 12:23:14.934079	[34mINFO[0m	Mounting BPF Filesystem at /sys/fs/bpf
2025-08-29 12:23:14.934173	[31mERROR[0m	Unable to mount BPF filesystem
github.com/kubearmor/KubeArmor/KubeArmor/log.Err
	/usr/src/KubeArmor/KubeArmor/log/logger.go:103
github.com/kubearmor/KubeArmor/KubeArmor/common.CheckOrMountBPFFs.func1
	/usr/src/KubeArmor/KubeArmor/common/bpffs.go:60
sync.(*Once).doSlow
	/usr/local/go/src/sync/once.go:78
sync.(*Once).Do
	/usr/local/go/src/sync/once.go:69
github.com/kubearmor/KubeArmor/KubeArmor/common.CheckOrMountBPFFs
	/usr/src/KubeArmor/KubeArmor/common/bpffs.go:58
github.com/kubearmor/KubeArmor/KubeArmor/monitor.NewSystemMonitor
	/usr/src/KubeArmor/KubeArmor/monitor/systemMonitor.go:211
github.com/kubearmor/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).InitSystemMonitor
	/usr/src/KubeArmor/KubeArmor/core/kubeArmor.go:258
github.com/kubearmor/KubeArmor/KubeArmor/core.KubeArmor
	/usr/src/KubeArmor/KubeArmor/core/kubeArmor.go:574
main.main
	/usr/src/KubeArmor/KubeArmor/main.go:63
runtime.main
	/usr/local/go/src/runtime/proc.go:283
2025-08-29 12:23:14.934984	[34mINFO[0m	Initializing eBPF system monitor
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1e3134b]

goroutine 1 [running]:
github.com/cilium/ebpf.(*Map).Update(0x23e5600?, {0x23e5600, 0xc0000ac880}, {0x268a920?, 0xc00003a0a0?}, 0x0)
	/go/pkg/mod/github.com/cilium/ebpf@v0.19.0/map.go:943 +0x2b
github.com/cilium/ebpf.(*Map).Put(...)
	/go/pkg/mod/github.com/cilium/ebpf@v0.19.0/map.go:938
github.com/kubearmor/KubeArmor/KubeArmor/monitor.(*SystemMonitor).UpdateNsKeyMap(0xc000125520, {0x26cab34?, 0x3?}, {0x0, 0x0}, {0x0, 0x0, 0x0, 0x0, 0x0})
	/usr/src/KubeArmor/KubeArmor/monitor/systemMonitor.go:363 +0x652
github.com/kubearmor/KubeArmor/KubeArmor/monitor.(*SystemMonitor).UpdateVisibility(0xc000125520)
	/usr/src/KubeArmor/KubeArmor/monitor/systemMonitor.go:465 +0x24b
github.com/kubearmor/KubeArmor/KubeArmor/monitor.(*SystemMonitor).initBPFMaps(0xc000125520)
	/usr/src/KubeArmor/KubeArmor/monitor/systemMonitor.go:232 +0x116
github.com/kubearmor/KubeArmor/KubeArmor/monitor.(*SystemMonitor).InitBPF(0xc000125520)
	/usr/src/KubeArmor/KubeArmor/monitor/systemMonitor.go:500 +0x1f8
github.com/kubearmor/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).InitSystemMonitor(0xc0001a0400)
	/usr/src/KubeArmor/KubeArmor/core/kubeArmor.go:263 +0x8a
github.com/kubearmor/KubeArmor/KubeArmor/core.KubeArmor()
	/usr/src/KubeArmor/KubeArmor/core/kubeArmor.go:574 +0x11f6
main.main()
	/usr/src/KubeArmor/KubeArmor/main.go:63 +0x3ed
```

--- a/deployments/helm/KubeArmor/templates/configmap.yaml
+++ b/deployments/helm/KubeArmor/templates/configmap.yaml
@@ -8,6 +8,7 @@ data:
   maxAlertPerSec: {{ .Values.kubearmorConfigMap.maxAlertPerSec | quote }}
   throttleSec: {{ .Values.kubearmorConfigMap.throttleSec | quote }}
   matchArgs: {{ .Values.kubearmorConfigMap.matchArgs | quote }}
+  karmor.yaml: | {{ toYaml .Values.kubearmorConfigMap | nindent 4 }}
 kind: ConfigMap
 metadata:
   labels:

--- a/deployments/helm/KubeArmor/templates/daemonset.yaml
+++ b/deployments/helm/KubeArmor/templates/daemonset.yaml
@@ -11,10 +11,11 @@ spec:
       kubearmor-app: kubearmor
   template:
     metadata:
-      annotations:
-        container.apparmor.security.beta.kubernetes.io/kubearmor: unconfined
       labels:
         kubearmor-app: kubearmor
+      {{- if eq .Values.environment.name "autopilot" }}
+        "cloud.google.com/matching-allowlist": accuknox-kubearmor-{{ include "chart.majorMinorVersion" . }}
+      {{- end }}
     spec:
       {{- if .Values.kubearmor.image.imagePullSecrets }}
       imagePullSecrets:
@@ -27,16 +28,21 @@ spec:
       containers:
       - args:
         - "-gRPC=32767"
-        - {{ printf "--tlsEnabled=%t" .Values.tls.enabled | quote }}
+        - "-procfsMount=/host/procfs"
+        - {{ printf "-tlsEnabled=%t" .Values.tls.enabled | quote }}
         {{- if .Values.tls.enabled }}
-        - {{ printf "--tlsCertPath=%s" .Values.kubearmor.tls.tlsCertPath | quote }}
-        - {{ printf "--tlsCertProvider=%s" .Values.kubearmor.tls.tlsCertProvider | quote }}
+        - {{ printf "-tlsCertPath=%s" .Values.kubearmor.tls.tlsCertPath | quote }}
+        - {{ printf "-tlsCertProvider=%s" .Values.kubearmor.tls.tlsCertProvider | quote }}
         {{- end }}
 
         {{- with .Values.kubearmor.args }}
         {{- toYaml . | nindent 8 }}
         {{- end }}
+        {{- if eq .Values.environment.name "autopilot" }}
+        image: {{printf "%s:%s" .Values.kubearmor.image.repository .Chart.Version }}
+        {{- else }}
         image: {{printf "%s:%s" .Values.kubearmor.image.repository .Values.kubearmor.image.tag}}
+        {{- end }}
         imagePullPolicy: {{ .Values.kubearmor.imagePullPolicy }}
         env:
         - name: KUBEARMOR_NODENAME
@@ -87,6 +93,8 @@ spec:
           {{- toYaml .Values.kubearmor.volumeMountsBottleRocket | trim | nindent 10 }}
         {{- else if eq .Values.environment.name "EKS" }}
           {{- toYaml .Values.kubearmor.volumeMountsEKS | trim | nindent 10 }}
+        {{- else if eq .Values.environment.name "autopilot" }}
+          {{- toYaml .Values.kubearmor.autoPilot.volumeMounts | trim | nindent 10 }}
         {{- else }} # generic
           {{- toYaml .Values.kubearmor.volumeMountsGeneric | trim | nindent 10 }}
         {{- end }}
@@ -96,10 +104,16 @@ spec:
         {{- end }}
       dnsPolicy: ClusterFirstWithHostNet
       hostNetwork: true
-      hostPID: true
+      securityContext:
+          appArmorProfile:
+            type: Unconfined
       initContainers:
-      - image: {{printf "%s:%s" .Values.kubearmorInit.image.repository .Values.kubearmorInit.image.tag}}
-        imagePullPolicy: {{ .Values.kubearmorInit.imagePullPolicy }}
+      - imagePullPolicy: {{ .Values.kubearmorInit.imagePullPolicy }}
+      {{- if eq .Values.environment.name "autopilot" }}
+        image: {{printf "%s:%s" .Values.kubearmorInit.image.repository .Chart.Version }}
+      {{- else }}
+        image: {{printf "%s:%s" .Values.kubearmorInit.image.repository .Values.kubearmorInit.image.tag}}
+      {{- end }}
         name: init
         securityContext:
           capabilities:
@@ -118,7 +132,11 @@ spec:
             - ALL
           privileged: false
         volumeMounts:
-        {{- toYaml .Values.kubearmor.initMounts | trim | nindent 10 }}
+        {{- if eq .Values.environment.name "autopilot" }}
+          {{- toYaml .Values.kubearmor.autoPilot.initMounts | trim | nindent 10 }}
+        {{- else }}  
+          {{- toYaml .Values.kubearmor.initMounts | trim | nindent 10 }}
+        {{- end }}
       nodeSelector:
         kubernetes.io/os: linux
       restartPolicy: Always
@@ -147,6 +165,8 @@ spec:
         {{- toYaml .Values.kubearmor.volumesBottleRocket | trim | nindent 8 }}
       {{- else if eq .Values.environment.name "EKS" }}
         {{- toYaml .Values.kubearmor.volumesEKS | trim | nindent 8 }}
+      {{- else if eq .Values.environment.name "autopilot" }}
+        {{- toYaml .Values.kubearmor.autoPilot.volumes | trim | nindent 8 }}
       {{- else }} # generic
         {{- toYaml .Values.kubearmor.volumesGeneric | trim | nindent 8 }}
       {{- end }}

--- a/deployments/helm/KubeArmor/templates/deployment.yaml
+++ b/deployments/helm/KubeArmor/templates/deployment.yaml
@@ -62,6 +62,7 @@ spec:
       {{- end }}
 {{- end }}
 ---
+{{- if ne .Values.environment.name "autopilot" }}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
@@ -77,7 +78,6 @@ spec:
   template:
     metadata:
       annotations:
-        container.apparmor.security.beta.kubernetes.io/manager: unconfined
         kubectl.kubernetes.io/default-container: manager
         kubearmor-policy: audited
       labels:
@@ -130,6 +130,7 @@ spec:
           readOnly: true
         - mountPath: /sys/kernel/security
           name: sys-path
+          readOnly: true
       securityContext:
         runAsNonRoot: true
       serviceAccountName: {{ .Values.kubearmorController.name }}
@@ -143,3 +144,4 @@ spec:
           path: /sys/kernel/security
           type: Directory
         name: sys-path
+{{- end }}

--- a/deployments/helm/KubeArmor/templates/helpers.tpl
+++ b/deployments/helm/KubeArmor/templates/helpers.tpl
@@ -1,0 +1,4 @@
+{{- define "chart.majorMinorVersion" -}}
+{{- $parts := splitList "." .Chart.Version -}}
+{{- printf "%s.%s" (index $parts 0) (index $parts 1) -}}
+{{- end -}}

--- a/deployments/helm/KubeArmor/values.yaml
+++ b/deployments/helm/KubeArmor/values.yaml
@@ -13,7 +13,7 @@ kubearmorRelay:
   enabled: true
   image:
     # kubearmor-init image repo
-    repository: kubearmor/kubearmor-relay-server
+    repository: docker.io/kubearmor/kubearmor-relay-server
     # kubearmor-init image tag
     tag: latest
     # Optional, but if there are a lot of image pulls required, Docker might be rate-limited. So, it's good to add pull secrets for production.
@@ -69,7 +69,7 @@ kubearmorRelay:
 kubearmorInit:
   image:
     # kubearmor-init image repo
-    repository: kubearmor/kubearmor-init
+    repository: docker.io/kubearmor/kubearmor-init
     # kubearmor-init image tag
     tag: stable
     # Optional, but if there are a lot of image pulls required, Docker might be rate-limited. So, it's good to add pull secrets for production.
@@ -94,7 +94,7 @@ kubearmorController:
   replicas: 1
   image:
     # kubearmor-controller image repo
-    repository: kubearmor/kubearmor-controller
+    repository: docker.io/kubearmor/kubearmor-controller
     # kubearmor-controller image tag
     tag: latest
     # Optional, but if there are a lot of image pulls required, Docker might be rate-limited. So, it's good to add pull secrets for production.
@@ -127,7 +127,7 @@ kubearmor:
   tolerations: ""
   image:
     # kubearmor daemonset image repo
-    repository: kubearmor/kubearmor
+    repository: docker.io/kubearmor/kubearmor
     # kubearmor daemonset image tag
     tag: stable
     # Optional, but if there are a lot of image pulls required, Docker might be rate-limited. So, it's good to add pull secrets for production.
@@ -168,6 +168,7 @@ kubearmor:
     - SYS_ADMIN
     - SYS_PTRACE
     - MAC_ADMIN
+    - NET_ADMIN
     - SYS_RESOURCE
     - IPC_LOCK
     - CAP_DAC_OVERRIDE
@@ -182,6 +183,51 @@ kubearmor:
   commonVolumes:
   - emptyDir: {}
     name: bpf
+
+  autoPilot:
+    volumeMounts:
+    - mountPath: /run/containerd/containerd.sock
+      name: containerd-socket
+      readOnly: true
+    - mountPath: /sys/kernel/debug
+      name: sys-kernel-debug-path
+    - mountPath: /host/procfs
+      name: proc-fs-mount
+      readOnly: true
+    - mountPath: /opt/kubearmor/karmor.yaml
+      name: kubearmor-config
+      subPath: karmor.yaml
+    initMounts:
+      - mountPath: /sys/kernel/debug
+        name: sys-kernel-debug-path
+      - mountPath: /host/procfs
+        name: proc-fs-mount
+        readOnly: true
+      - mountPath: /opt/kubearmor/karmor.yaml
+        name: kubearmor-config
+        subPath: karmor.yaml
+      - mountPath: /opt/kubearmor/BPF
+        name: bpf
+    volumes:
+      - hostPath:
+          path: /run/containerd/containerd.sock
+          type: Socket
+        name: containerd-socket
+      - hostPath:
+          path: /sys/kernel/debug
+          type: Directory
+        name: sys-kernel-debug-path
+      - hostPath:
+          path: /proc
+          type: Directory
+        name: proc-fs-mount
+      - configMap:
+          defaultMode: 420
+          items:
+          - key: karmor.yaml
+            path: karmor.yaml
+          name: kubearmor-config
+        name: kubearmor-config
 
   initMounts:
   - mountPath: /opt/kubearmor/BPF

--- a/getting-started/deployment_guide.md
+++ b/getting-started/deployment_guide.md
@@ -2,6 +2,8 @@
 
 This guide assumes you have access to a [k8s cluster](support_matrix.md). If you want to try non-k8s mode, for instance systemd mode to protect/audit containers or processes on VMs/bare-metal, check [here](kubearmor_vm.md).
 
+If you're deploying kubearmor on GKE autopilot follow the instructions provide with [GKE Autopilot Installation Guide](kubearmor_installation_on_gke_autopilot.md)
+
 Check the [KubeArmor support matrix](support_matrix.md) to verify if your platform is supported.
 
 ## Install KubeArmor

--- a/getting-started/kubearmor_installation_on_gke_autopilot.md
+++ b/getting-started/kubearmor_installation_on_gke_autopilot.md
@@ -1,0 +1,42 @@
+# Install KubeArmor on GKE Autopilot
+
+GKE autopilot by default restrict the workloads that requires advance permissions/capabilities i.e. `hostPath` mounts, capabilities like `SYS_ADMIN`, etc. User can deploy such applications/workloads using a feature called Workload Allowlist.
+
+KubeArmor maintains Allowlist for every stable release as part of [GKE partner workloads](https://cloud.google.com/kubernetes-engine/docs/how-to/run-autopilot-partner-workloads). Users can use the allowlist to deploy KubeArmor on their GKE cluster.
+
+### 1. ⬇️ Pull KubeArmor Allowlist 
+
+Apply the following yaml to sync kubearmor allowlist from GKE partner repository
+
+```
+# kubearmor-allowlist.yaml
+apiVersion: auto.gke.io/v1
+kind: AllowlistSynchronizer
+metadata:
+  name: kubearmor-allow-list
+spec:
+  allowlistPaths:
+  - Accuknox/kubearmor/*
+
+```
+verify that kubearmor allowlist(s) are synchronized
+
+```
+kubectl get WorkloadAllowlist
+```
+
+### 2. 🚀 Deploy KubeArmor using Helm Chart
+
+```
+helm repo add kubearmor https://kubearmor.github.io/charts
+helm repo update kubearmor
+helm upgrade --install kubearmor kubearmor/kubearmor \
+  --set environment.name=autopilot \
+  -n kubearmor --create-namespace
+```
+
+### 3. 🎉 Verify KubeArmor is running
+
+```
+kubectl get pods -n kubearmor
+```

